Library/Homebrew/test/test_diagnostic.rb

```diff
@@ -142,7 +142,7 @@ def test_check_user_path_bin
   def test_check_user_path_sbin
     sbin = HOMEBREW_PREFIX/"sbin"
     ENV["PATH"] = "#{HOMEBREW_PREFIX}/bin#{File::PATH_SEPARATOR}" +
-      ENV["PATH"].gsub(%r{(?:^|#{File::PATH_SEPARATOR})#{sbin}}, "")
+                  ENV["PATH"].gsub(/(?:^|File::PATH_SEPARATOR.to_s)sbin.to_s/, "")
```

Comment:
The substitution of `#{File::PATH_SEPARATOR}` with `File::PATH_SEPARATOR.to_s` is wrong and the same applies to the `#{sbin}` and `sbin.to_s` pair. A relatively easy way to check this is to enter both regular expressions into `brew irb` (though we'll need to also provide `sbin`):

```console
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> sbin = HOMEBREW_PREFIX/"sbin"
=> #<Pathname:/opt/homebrew/sbin>
irb(main):002:0> %r{(?:^|#{File::PATH_SEPARATOR})#{sbin}}
=> /(?:^|:)\/opt\/homebrew\/sbin/
irb(main):003:0> /(?:^|File::PATH_SEPARATOR.to_s)sbin.to_s/
=> /(?:^|File::PATH_SEPARATOR.to_s)sbin.to_s/
```

The correct substitution should be:

```ruby
/(?:^|#{Regexp.escape(File::PATH_SEPARATOR)})#{Regexp.escape(sbin)}/
```

(`Regexp.escape` makes additionally sure to properly handle any characters that we want to be interpreted literally, but that have special meaning when used inside a regular expression.)